test/rules/no-access-state-in-setstate/test.tsx.lint

```diff
@@ -0,0 +1,68 @@
+class SomeReactComponent extends React.Component {
+
+    someClassFunction() {
+
+        this.fooBar({
+            foo: this.state.foo
+        });
+
+        this.setState({
+            foo: "foo",
+            bar: this.barz
+        });
+
+        this.setState(
+            {
+                foo: "foo"
+            },
+            () => this.fooBar(this.state.foo);
+        );
+
+        this.setState(prevState => ({
+            foo: !prevState.foo
+        }));
+
+        this.setState((prevState, currentProps) => ({
+            foo: !prevState.foo,
+            bar: currentProps.bar
+        }));
+
+        this.setState({
+        ~~~~~~~~~~~~~~~
+            foo: !this.state.foo
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        });
+~~~~~~~~~~ [0]
+
+        this.setState({
+        ~~~~~~~~~~~~~~~
+            foo: this.fooBar(this.state.foo)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        });
+~~~~~~~~~~ [0]
+
+        this.setState((prevState, currentProps) => ({
+        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+            foo: !this.state.foo,
```

Comment:
it would of course be much nicer to have the marker only on the occurrence(s) of `this.state` but I would consider it nice to have.

Reply:
I sat down again with my code and now I'm pretty sure I found a still very easy and even more reliable way to not only find accesses of this.state but also can now display the error message exactly at the right node.
I will push the changes at the weekend! Thanks for the idea!

Reply:
Or even better push it right away. Looking forward to your review!
I now first check if a node is a setState call and if so, I check each child of the first argument of that call, determining if that node is a this.state access.